src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts

```diff
@@ -236,18 +240,41 @@ export class GeneComparisonToolComponent implements OnInit, AVI, OnDestroy {
   getGeneProperty(gene: GCTGene, property: string) {
     return property.split('.').reduce((o: any, i: any) => o[i], gene);
   }
-
+  
   initData(genes: GCTGene[]) {
+    // hide brain region columns initially
     this.brainRegionsColumns.forEach(c => c.visible = false);
-
+    
     const pinnedGenes: GCTGene[] = [];
-    const currentPinnedGenesCache = this.getPinnedGenesCache(
-      this.category,
-      this.subCategory
-    );
-    const urlPins = currentPinnedGenesCache.length
-      ? currentPinnedGenesCache.map((g: GCTGene) => g.uid)
-      : this.getUrlParam('pinned', true);
+    
+    const currentPinnedGenesCache = this.pinnedGenesCache;
+
+    let previousPins: (string | undefined)[] = [];
+
+    if (this.currentCategoriesMatchLastPinnedCategories()) {
+      // no need to load from cache if it was last pinned from the current page
+      // previousPins = this.getUrlParam('pinned', true);
+      previousPins = currentPinnedGenesCache.map((g: GCTGene) => g.uid);
+    } else {
+      // if it is the initial load, the categories will not match since the last pinned categories 
+      // will be empty strings.  In this case we want to check the url to see if this was a shared url
+      // with pinned genes/proteins.
+      
+      // since the last pinned category is blank, we don't need to check the subcategory
+      if (this.lastPinnedCategory === '') {
+        // check the url for pinned genes/proteins
+        previousPins = this.getUrlParam('pinned', true);
+        this.setLastPinnedCategories();
+      } else {
+        // categories don't match, so grab it from the cache
+        if (this.category === 'RNA - Differential Expression') {
+          previousPins = currentPinnedGenesCache
+            .map((g: GCTGene) => g.uid?.substring(0, 15));
```

Comment:
Could just get the ensembl gene id directly instead? 

```suggestion
          previousPins = currentPinnedGenesCache
            .map((g: GCTGene) => g.ensembl_gene_id;
```

Reply:
you are right